### PR TITLE
Calculate full BC artifact URL when updating it

### PR DESCRIPTION
**Pull request by mazhelez**
Calculating BC artifact might switch between _bcinsider_ and _bcartifacts_ (typically around branching when a new version is available but not yet released). This PR addresses the issue when the storage account needs to be changed as well.

[AB#524129](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/524129)

### CODE REVIEW

File: build/scripts/UpdateBCArtifactVersion.ps1 [hunk 1]
```diff
@@ -26,25 +26,18 @@ Import-Module $PSScriptRoot\EnlistmentHelperFunctions.psm1
...
-    Write-Host "Latest BCArtifact version: $latestArtifactVersion"
+    $latestArtifactUrl = Get-LatestBCArtifactUrl -minimumVersion $currentVersion
 
-    if($latestArtifactVersion -gt $currentArtifactVersion) {
-        Write-Host "Updating BCArtifact version from $currentArtifactVersion to $latestArtifactVersion"
+    Write-Host "Latest BCArtifact URL: $latestArtifactUrl"
 
-        $artifactValue = $artifactValue -replace $currentArtifactVersion, $latestArtifactVersion
-        Set-ConfigValue -Key "artifact" -Value $artifactValue -ConfigType AL-Go
+    if($latestArtifactUrl -ne $currentArtifactVersion) {
```
Comment (aholstrup1) [inline build/scripts/UpdateBCArtifactVersion.ps1]:
```suggestion
    if($latestArtifactUrl -ne $currentArtifactUrl) {
```

**Review by aholstrup1 — APPROVED** [review]
LGTM - Maybe just test it in your fork if you can first. Then.. Ship ship ship 🚢

File: build/scripts/UpdateBCArtifactVersion.ps1 (hunk 1 above)
Comment (mazhelez) [reply build/scripts/UpdateBCArtifactVersion.ps1]: Nice catch! Thanks!

**Comment by mazhelez** [comment]
> LGTM - Maybe just test it in your fork if you can first. Then.. Ship ship ship 🚢

Seems to work: https://github.com/mazhelez/BCApps/pull/65/files

**Review by aholstrup1 — APPROVED** [review]

**Review by elbardel — APPROVED** [review]

**Pull request merged**